pkg/volume/vsphere_volume/vsphere_volume_util.go

```diff
@@ -73,15 +82,56 @@ func (util *VsphereDiskUtil) CreateVolume(v *vsphereVolumeProvisioner) (vmDiskPa
 	// the values to the cloud provider.
 	for parameter, value := range v.options.Parameters {
 		switch strings.ToLower(parameter) {
-		case "diskformat":
+		case diskformat:
 			volumeOptions.DiskFormat = value
 		case "datastore":
 			volumeOptions.Datastore = value
+		case Policy_HostFailuresToTolerate:
```

Comment:
We would creating so many methods for all the VSAN capabilities. This way it is easy to understand as for each case it would validate the  VSAN capability and concatenate it to volumeOptions.StorageProfileData.